### ISSUE TITLE
Preview button does not show full size images

### ISSUE THREAD

**Issue by thelittlefireman** [lukaspieper/Gcam-Services-Provider#11]
**Describe the bug**
i've got a crash with gcam 7.6 (and previous 7.5) when i it the button "show picture"

**To Reproduce**
1. open the gcam app
2. take a picture
3. try directly to see it by clicking on the "picture" button
4. Gcam restart on camera activity

**Expected behavior**

**Log**
```
11-04 19:17:19.930 12693 12919 E native  : lens_offset.cc:119 timestamp 904844344241 invalid. (-555583535 ns away from the latest one)
11-04 19:17:19.930 12693 12919 E native  : lens_offset.cc:119 timestamp 904846344241 invalid. (-553583535 ns away from the latest one)
11-04 19:17:19.930 12693 12919 E native  : lens_offset.cc:119 timestamp 904848344241 invalid. (-551583535 ns away from the latest one)
11-04 19:17:19.930 12693 12919 W GyroTransformCalc: Previous or current projection matrix cannot be computed. Defaulting to identity
11-04 19:17:19.930 12693 12919 I chatty  : uid=10117(com.google.android.GoogleCamera) mv-gyro-exec identical 10 lines
11-04 19:17:19.930 12693 12919 W GyroTransformCalc: Previous or current projection matrix cannot be computed. Defaulting to identity
11-04 19:17:20.464   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 27.027027
11-04 19:17:20.599 12693 13884 W JankMonitorFacade: PHOTO > result sensor delay = 1,51 > 1,10
11-04 19:17:20.653 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,34 > 0,25
11-04 19:17:20.854 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,33 > 0,25
11-04 19:17:21.175   881  7531 I QC2Comp : [avcD_15] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.050/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:21.175   881  7528 I QC2Comp : [avcD_14] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.050/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:21.228   858   965 D VSC     : @ 919.638: [WO] tilt angle 44
11-04 19:17:21.228   858   965 D VSC     : @ 919.638: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:21.228   858   965 D VSC     : @ 919.638: [WO] orientation angle 343, orientation 4
11-04 19:17:21.228   858   965 D VSC     : @ 919.638: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:21.474   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 25.742574
11-04 19:17:21.736   881  7535 W C2BqBuffer: last successful dequeue was 594962297 us ago, 47570 consecutive failures
11-04 19:17:21.820   881  7537 W C2BqBuffer: last successful dequeue was 594898744 us ago, 47562 consecutive failures
11-04 19:17:22.479   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.875622
11-04 19:17:23.255   858   965 D VSC     : @ 921.665: [WO] tilt angle 45
11-04 19:17:23.255   858   965 D VSC     : @ 921.665: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:23.255   858   965 D VSC     : @ 921.665: [WO] orientation angle 342, orientation 4
11-04 19:17:23.255   858   965 D VSC     : @ 921.665: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:23.483   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.900398
11-04 19:17:23.755   858   965 D CHRE    : @ 922.165: [ImuCal] Dynamic sensor configuration: stand-by.
11-04 19:17:24.487   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.900398
11-04 19:17:24.489   881 13935 I QC2Comp : [avcE_17] Stats: Works: Q(1)/Done(1)| Work-Rate: Q(0.2/s Avg=0.2/s) Done(0.197/s Avg=0.197/s)| Stream: 30.00fps 0.0Kbps| Pending(0) i/p-done(0)
11-04 19:17:24.501   881  2137 D BufferPoolAccessor2.0: bufferpool2 0xd5ac0808 : 5(3932160 size) total buffers - 4(3145728 size) used buffers - 0/5 (recycle/alloc) - 1/1 (fetch/transfer)
11-04 19:17:24.501   881  2137 D BufferPoolAccessor2.0: evictor expired: 1, evicted: 1
11-04 19:17:24.644 12693 12844 D BufferPoolAccessor2.0: bufferpool2 0x523afa3c28 : 5(38400 size) total buffers - 4(30720 size) used buffers - 0/5 (recycle/alloc) - 0/0 (fetch/transfer)
11-04 19:17:24.644 12693 12844 D BufferPoolAccessor2.0: evictor expired: 1, evicted: 1
11-04 19:17:24.647  1225  2141 D BufferPoolAccessor2.0: evictor expired: 4, evicted: 0
11-04 19:17:25.282   858   965 D VSC     : @ 923.691: [WO] tilt angle 45
11-04 19:17:25.282   858   965 D VSC     : @ 923.692: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:25.282   858   965 D VSC     : @ 923.692: [WO] orientation angle 342, orientation 4
11-04 19:17:25.282   858   965 D VSC     : @ 923.692: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:25.492   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.875622
11-04 19:17:25.501   881  2137 D BufferPoolAccessor2.0: evictor expired: 1, evicted: 0
11-04 19:17:25.645 12693 12844 D BufferPoolAccessor2.0: evictor expired: 4, evicted: 0
11-04 19:17:25.647  1225  2141 D BufferPoolAccessor2.0: bufferpool2 0x51dfc09a28 : 0(0 size) total buffers - 0(0 size) used buffers - 0/0 (recycle/alloc) - 0/0 (fetch/transfer)
11-04 19:17:25.647  1225  2141 D BufferPoolAccessor2.0: evictor expired: 1, evicted: 1
11-04 19:17:26.176   881  7528 I QC2Comp : [avcD_14] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.050/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:26.176   881  7531 I QC2Comp : [avcD_15] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.050/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:26.393 12693 13884 W JankMonitorFacade: PHOTO > result sensor delay = 1,51 > 1,10
11-04 19:17:26.424 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,34 > 0,25
11-04 19:17:26.517   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 25.365854
11-04 19:17:26.557 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,30 > 0,25
11-04 19:17:26.557 12693 13884 W JankMonitorFacade: PHOTO > result sensor delay = 1,30 > 1,10
11-04 19:17:26.737   881  7535 W C2BqBuffer: last successful dequeue was 599964066 us ago, 48043 consecutive failures
11-04 19:17:26.821   881  7537 W C2BqBuffer: last successful dequeue was 599899725 us ago, 48035 consecutive failures
11-04 19:17:26.999 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,39 > 0,25
11-04 19:17:27.036 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,99 > 0,25
11-04 19:17:27.082  1352  1554 I GnssLocationProvider: GNSS HAL Requesting location updates from network provider for 10000 millis.
11-04 19:17:27.084  1352  1554 W GnssLocationProvider: Unable to request location.
11-04 19:17:27.084  1352  1554 W GnssLocationProvider: java.lang.IllegalArgumentException: provider doesn't exist: network
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at com.android.server.location.LocationManagerService.requestLocationUpdatesLocked(LocationManagerService.java:1946)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at com.android.server.location.LocationManagerService.requestLocationUpdates(LocationManagerService.java:1929)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1179)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1140)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at com.android.server.location.gnss.GnssLocationProvider.handleRequestLocation(GnssLocationProvider.java:828)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at com.android.server.location.gnss.GnssLocationProvider.access$2700(GnssLocationProvider.java:100)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at com.android.server.location.gnss.GnssLocationProvider$ProviderHandler.handleMessage(GnssLocationProvider.java:2073)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at android.os.Handler.dispatchMessage(Handler.java:106)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at android.os.Looper.loop(Looper.java:223)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at android.os.HandlerThread.run(HandlerThread.java:67)
11-04 19:17:27.084  1352  1554 W GnssLocationProvider:  at com.android.server.ServiceThread.run(ServiceThread.java:44)
11-04 19:17:27.308   858   965 D VSC     : @ 925.718: [WO] tilt angle 43
11-04 19:17:27.308   858   965 D VSC     : @ 925.718: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:27.308   858   965 D VSC     : @ 925.718: [WO] orientation angle 342, orientation 4
11-04 19:17:27.308   858   965 D VSC     : @ 925.718: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher: Photos launch was cancelled
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher: java.util.concurrent.CancellationException: Task was cancelled.
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at poc.a(PG:132)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at poc.get(PG:82)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at pqt.a(PG:23)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at ppp.a(PG:16)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at ppk.run(PG:5)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at poz.execute(PG:3)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at poc.b(PG:77)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at poc.a(PG:76)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at poc.cancel(PG:52)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at dff.h(PG:23)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at dfo.a(PG:128)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at dfo.b(PG:25)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at dfr.onAnimationEnd(PG:16)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:554)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.animation.ValueAnimator.endAnimation(ValueAnimator.java:1250)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1492)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.animation.AnimationHandler.access$100(AnimationHandler.java:37)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.view.Choreographer$CallbackRecord.run(Choreographer.java:970)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.view.Choreographer.doCallbacks(Choreographer.java:796)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.view.Choreographer.doFrame(Choreographer.java:727)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:957)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.os.Handler.handleCallback(Handler.java:938)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.os.Handler.dispatchMessage(Handler.java:99)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.os.Looper.loop(Looper.java:223)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at android.app.ActivityThread.main(ActivityThread.java:7656)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at java.lang.reflect.Method.invoke(Native Method)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at com.android.internal.os.ExecInit.main(ExecInit.java:43)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
11-04 19:17:27.336 12693 12693 W CAM_Photos1UpLauncher:         at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:399)
11-04 19:17:27.431 12693 12725 D skia    : libjpeg error 116 <Corrupt JPEG data: 18 extraneous bytes before marker 0xdb> from output_message
11-04 19:17:27.545   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 27.237354
11-04 19:17:28.549   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.900398
11-04 19:17:29.335   858   965 D VSC     : @ 927.745: [WO] tilt angle 46
11-04 19:17:29.336   858   965 D VSC     : @ 927.745: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:29.336   858   965 D VSC     : @ 927.745: [WO] orientation angle 340, orientation 4
11-04 19:17:29.336   858   965 D VSC     : @ 927.745: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:29.490   881 13935 I QC2Comp : [avcE_17] Stats: Works: Q(1)/Done(1)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.099/s)| Stream: 30.00fps 0.0Kbps| Pending(0) i/p-done(0)
11-04 19:17:29.554   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.875622
11-04 19:17:30.558   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.900398
11-04 19:17:31.176   881  7528 I QC2Comp : [avcD_14] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.050/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:31.176   881  7531 I QC2Comp : [avcD_15] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.050/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:31.362   858   965 D VSC     : @ 929.772: [WO] tilt angle 58
11-04 19:17:31.363   858   965 D VSC     : @ 929.772: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:31.363   858   965 D VSC     : @ 929.772: [WO] orientation angle 345, orientation 4
11-04 19:17:31.363   858   965 D VSC     : @ 929.772: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:31.388   858   965 D CHRE    : @ 929.797: [ImuCal] Dynamic sensor configuration: stand-by.
11-04 19:17:31.563   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 24.875622
11-04 19:17:31.738   881  7535 W C2BqBuffer: last successful dequeue was 604964661 us ago, 48515 consecutive failures
11-04 19:17:31.823   881  7537 W C2BqBuffer: last successful dequeue was 604901069 us ago, 48507 consecutive failures
11-04 19:17:31.938 12693 13884 W JankMonitorFacade: PHOTO > result sensor delay = 1,51 > 1,10
11-04 19:17:31.993 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,34 > 0,25
11-04 19:17:32.565   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 27.944112
11-04 19:17:32.784 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,33 > 0,25
11-04 19:17:32.905 12693 13884 W JankMonitorFacade: PHOTO > result sensor delay = 1,51 > 1,10
11-04 19:17:32.963 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,34 > 0,25
11-04 19:17:33.389   858   965 D VSC     : @ 931.798: [WO] tilt angle 61
11-04 19:17:33.389   858   965 D VSC     : @ 931.798: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:33.389   858   965 D VSC     : @ 931.798: [WO] orientation angle 349, orientation 4
11-04 19:17:33.389   858   965 D VSC     : @ 931.798: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:33.599   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 29.013540
11-04 19:17:34.490   881 13935 I QC2Comp : [avcE_17] Stats: Works: Q(1)/Done(1)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.066/s)| Stream: 30.00fps 0.0Kbps| Pending(0) i/p-done(0)
11-04 19:17:34.631   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 30.038760
11-04 19:17:35.416   858   965 D VSC     : @ 933.826: [WO] tilt angle 62
11-04 19:17:35.416   858   965 D VSC     : @ 933.826: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:35.416   858   965 D VSC     : @ 933.826: [WO] orientation angle 349, orientation 4
11-04 19:17:35.417   858   965 D VSC     : @ 933.826: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:35.665   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 30.009681
11-04 19:17:36.176   881  7528 I QC2Comp : [avcD_14] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.049/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:36.176   881  7531 I QC2Comp : [avcD_15] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.049/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:36.698   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 30.009681
11-04 19:17:36.744   881  7535 W C2BqBuffer: last successful dequeue was 609970474 us ago, 48986 consecutive failures
11-04 19:17:36.829   881  7537 W C2BqBuffer: last successful dequeue was 609906925 us ago, 48978 consecutive failures
11-04 19:17:37.443   858   965 D VSC     : @ 935.853: [WO] tilt angle 62
11-04 19:17:37.443   858   965 D VSC     : @ 935.853: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:37.443   858   965 D VSC     : @ 935.853: [WO] orientation angle 348, orientation 4
11-04 19:17:37.443   858   965 D VSC     : @ 935.853: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:37.731   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 29.980658
11-04 19:17:38.763   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 30.038760
11-04 19:17:39.470   858   965 D VSC     : @ 937.880: [WO] tilt angle 62
11-04 19:17:39.470   858   965 D VSC     : @ 937.880: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:39.470   858   965 D VSC     : @ 937.880: [WO] orientation angle 348, orientation 4
11-04 19:17:39.470   858   965 D VSC     : @ 937.880: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:39.491   881 13935 I QC2Comp : [avcE_17] Stats: Works: Q(1)/Done(1)| Work-Rate: Q(0.0/s Avg=0.0/s) Done(0.000/s Avg=0.050/s)| Stream: 30.00fps 0.0Kbps| Pending(0) i/p-done(0)
11-04 19:17:39.796   834  1053 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 30.009681
11-04 19:17:40.270 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 1,00 > 0,25
11-04 19:17:40.271 12693 13884 W JankMonitorFacade: PHOTO > result sensor delay = 2,00 > 1,10
11-04 19:17:40.317 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,50 > 0,25
11-04 19:17:40.464   842   842 I android.hardware.health@2.0-service.bonito: capacity: 3100000
11-04 19:17:40.466   842   842 E BatteryDefender: Failed to read /sys/class/power_supply/wireless/online
11-04 19:17:40.644  1352  4511 W ActivityManager: Foreground service started from background can not have location/camera/microphone access: service au.com.vodafone.dreamlabapp/.workflow.calculation.NotificationService
11-04 19:17:40.829   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 29.041626
11-04 19:17:40.857  2106  2106 D InterruptionStateProvider: No bubble up: not allowed to bubble: 0|au.com.vodafone.dreamlabapp|41129|null|10132
11-04 19:17:40.858  2106  2106 D InterruptionStateProvider: No heads up: unimportant notification: 0|au.com.vodafone.dreamlabapp|41129|null|10132
11-04 19:17:41.177   881  7528 I QC2Comp : [avcD_14] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.049/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:41.177   881  7531 I QC2Comp : [avcD_15] Stats: Works: Q(34)/Done(30)| Work-Rate: Q(0.0/s Avg=0.1/s) Done(0.000/s Avg=0.049/s)| Stream: 25.00fps 1.9Mbps| Pending(4) i/p-done(2)
11-04 19:17:41.498   858   965 D VSC     : @ 939.907: [WO] tilt angle 62
11-04 19:17:41.498   858   965 D VSC     : @ 939.907: [WO] isFlat() nearest_rotation: 4, flat_angle: 65
11-04 19:17:41.498   858   965 D VSC     : @ 939.907: [WO] orientation angle 348, orientation 4
11-04 19:17:41.498   858   965 D VSC     : @ 939.907: [WO] rejected by isOrientationAngleAcceptable
11-04 19:17:41.754   881  7535 W C2BqBuffer: last successful dequeue was 614980638 us ago, 49457 consecutive failures
11-04 19:17:41.838   881  7537 W C2BqBuffer: last successful dequeue was 614916691 us ago, 49449 consecutive failures
11-04 19:17:41.862   834  1052 I ISPHVX  : PREVIEW_FRAMES_PER_SECOND: 30.009681
11-04 19:17:42.083 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,33 > 0,25
11-04 19:17:42.324 12693 13884 W JankMonitorFacade: PHOTO > result sensor delay = 1,50 > 1,10
11-04 19:17:42.369 12693 13884 W JankMonitorFacade: PHOTO > rel Δ(result sensor timestamp) = 0,33 > 0,2
```

**Screenshots**
![](https://i.imgur.com/t99CR1k.jpeg)

**Smartphone (please complete the following information, the more detailed, the better)**
 - Device:  Pixel 3a
 - OS/ROM: Android 11, GrapheneOS 11
 - Google Camera version (original): original 7.5.x, original 7.6.x, download from apkpure
 - Gcam-Services-Provider version: 1.1 (from gthub release)

**Additional context**

Thanks

**Comment by lukaspieper** [comment]
Hi,

I'm running GrapheneOS 11 with Gcam 7.5. Since you write that you had the same problem with version 7.5, I must say that I cannot reproduce the issue.

In the official Gcam app, the button that takes you to the picture gallery app is hard-wired to Google photos app. If the app is not installed, Gcam will run an animation without crashing at least on my device.

I like that you have posted a log, but unfortunately this doesn't help me either, because the real exception occurs in a task and is swallowed.

Furthermore, the log level of the exception is warning, which does not sound like the cause of a crash to me, but this is just speculation on my part.

By the way, I have created an [issue template](https://github.com/lukaspieper/Gcam-Services-Provider/blob/master/.github/ISSUE_TEMPLATE/bug-or-crash-report.md) for bugs, it would be great if you adapt your issues to it or share more information.

**Comment by thelittlefireman** [comment]
hi i'm on GrapheneOS 11 too. Could you share me links or informations about where you are downloading gcam apk , please ?
Is there anything i can do to get you more usefull logs ?

**Comment by lukaspieper** [comment]
I download gcam apks from apkmirror.

You could record the crash (gif/video). Because I'm not sure if we define "crash" the same way. 

If you click on the button, will Gcam crash, so that you are thrown back into the launcher? Possibly with the system dialog saying that Gcam crashed?

**Comment by innir** [comment]
Did you try if this (https://github.com/lukaspieper/Gcam-Services-Provider/issues/6#issuecomment-742816829) fix helps here too?

**Comment by firestorm99** [comment]
gcam.7.4.201
gcam service provider 1.1
grapheneos pixel 4a build RQ1A.201205.008.2020.12.08.08

taking pictures works, viewing them inside the app does not, the animation does not complete. to see the cam feed again you need to switch the mode once.

**Comment by innir** [comment]
> taking pictures works, viewing them inside the app does not, the animation does not complete. to see the cam feed again you need to switch the mode once.

Have you installed Google Photos, it will not work without.

**Comment by firestorm99** [comment]
> > taking pictures works, viewing them inside the app does not, the animation does not complete. to see the cam feed again you need to switch the mode once.
> 
> Have you installed Google Photos, it will not work without.

that did the trick :-)

**Comment by lukaspieper** [comment]
Alternatively, there are also modded Gcam versions that work with any gallery app, but everyone has to decide for themselves if they want to use a modded version.

I suspect that @thelittlefireman  was referring to the same problem. Technically, though, I wouldn't call it a crash, it's more of a non-functioning feature.

It would be interesting to know how Gcam and Google Photos are "wired".

**Comment by thelittlefireman** [comment]
> Alternatively, there are also modded Gcam versions that work with any gallery app, but everyone has to decide for themselves if they want to use a modded version.
> 
> I suspect that @thelittlefireman  was referring to the same problem. Technically, though, I wouldn't call it a crash, it's more of a non-functioning feature.
> 
> It would be interesting to know how Gcam and Google Photos are "wired".

Yes, that's exactly the same problem. I haven't test with Google photo yet. I Will let you know. By the way, i agree that would be intresting to not be forced to install Google photo cause i think if People are using this gcam they would like to avoid using Google services/tracking.

Maybe it's Just a simple ipc/binder service exposed by Google photo ?

**Comment by lukaspieper** [comment]
I tried around a bit (see [feature branch](https://github.com/lukaspieper/Gcam-Services-Provider/tree/feature/show-images-button)). I have created an intent which, except for the extras, corresponds to a captured intent from Gcam. My intent also opens Google Photos. 

Furthermore I created a stub app (of Google Photos) that also reacts to my intent. Only problem: Gcam does not use the stub app. Maybe I missed something simple or Gcam does interact with Google Photos in a different way. 

Perhaps someone of you would like to take a look at the feature branch and experiment with it further.

**Comment by tifriis208** [comment]
For information, calyx os release an app to solve this :  https://calyxos.org/news/2021/09/30/gcam-photos-preview/

**Comment by lukaspieper** [comment]
> For information, calyx os release an app to solve this : https://calyxos.org/news/2021/09/30/gcam-photos-preview/

Thanks a lot for the hint. Inspired by the solution from Calyx, I added a redirect to the default gallery app. I don't want to implement and maintain a photos viewer like Calyx did. Would be great if you would try it. You can get an APK from [here](https://github.com/lukaspieper/Gcam-Services-Provider/actions/runs/1295325750). Inside the zip take the APK with "photos" in the name.

Please note that you need to uninstall GCam-Services-Provider manually before and if you have Google Photos installed you also need to remove that.

**Comment by gryban1** [comment]
Worked for a short duration* on my device (Pixel 5, GrapheneOS, GCam 8.2.400)!

*maybe due to https://github.com/lukaspieper/Gcam-Services-Provider/discussions/52 where the gcam 8.2.400 seems to need sth only google photos provides.

**Comment by lukaspieper** [comment]
[Fixed with version 1.3](https://github.com/lukaspieper/Gcam-Services-Provider/releases/tag/v1.3)

@gryban1 Thanks for testing. I'm sorry but I still have no new information regarding #52.

**Comment by gryban1** [comment]
@lukaspieper No problem. Thanks for your good work done here with the whole project. :)
